viewing_party/party.py

```diff
@@ -1,23 +1,176 @@
 # ------------- WAVE 1 --------------------
 
+from shutil import move
+
+
 def create_movie(title, genre, rating):
-    pass
+    movie_info = [title, genre, rating]
+    if None in movie_info:
+        movie = None
+    else:
+        movie = {"title": title, "genre": genre, "rating": rating}
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
 
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data,movie_title):
+    for i in range(len(user_data["watchlist"])):
+        for info in user_data["watchlist"][i].values():
+            if info == movie_title:
+                # Remove movie from watchlist
+                watched_movie = user_data["watchlist"].pop(i)
+                add_to_watched(user_data, watched_movie)
+    
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    if len(user_data["watched"]) == 0:
+        return float(0)
+
+    rating_sum = 0
+    for i in range(len(user_data["watched"])):
+        rating_sum += user_data["watched"][i]["rating"]
+    
+    avg_rating = rating_sum / len(user_data["watched"])
+    return avg_rating
 
+def get_most_watched_genre(user_data):
+    if len(user_data["watched"]) == 0:
+        return None
+
+    watched_genres = {}
+    for movie in user_data["watched"]:
+        movie_genre = movie["genre"]
+        if movie_genre in watched_genres:
+            watched_genres[movie_genre] += 1
+        else:
+            watched_genres[movie_genre] = 1
+
+    most_watched = max(watched_genres, key = watched_genres.get)
+    return most_watched
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_movie_titles(watched_list):
+    """
+    Takes in one parameter, a list of dictionaries
+    """
+    movie_titles = []
+
+    for movies in watched_list:
+        for info in movies.keys():
+            if info == "title":
+                movie_titles.append(movies["title"])

```

Comment:
I think this is overkill for this helper function! We can access the "title" key inside the first loop. Also, remember what the data you are iterating over contains. The iterator is a single `movie`

```suggestion
    for movie in watched_list:
         movie_titles.append(movie["title"])
```